src/priority-queue-original.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * A priority queue stores a list of items but each can have a numeric priority value.
+ * Items with a higher priority are dequeued before items with a lower priority.
+ * Implemented as a hash of arrays where the hash keys are priority values.
+ */
+function PriorityQueue(size) {
+	this.store = {};	// keys are priorities, values are arrays of elements
+	this.count = 0;
+
+	// adds an item
+	// priority must be an integer (higher value has higher priority)
```

Comment:
```suggestion
/**                                                                    
 * Adds a +value+ to this queue of the given +priority+. The higher
 * the number given, the higher the priority.
 *                                                                     
 * @param {Any} value
 * @param {Integer} priority
 */
```
I recommend using [JSDoc](https://jsdoc.app/about-getting-started.html) style comment blocks like this. Most every editor/IDE has either built-in or plugin support for these. See [VS Code here](https://code.visualstudio.com/Docs/languages/javascript#_jsdoc-support), or [vim here](https://github.com/kkoomen/vim-doge#readme).